### PR TITLE
Update service host with latest 3.x service fabric packages and build for netcoreapp3.1

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

File: src/Microsoft.DotNet.ServiceFabric.ServiceHost/EnvironmentConfigMapper.cs [hunk 1]
```diff
@@ -17,20 +17,20 @@ public EnvironmentConfigMapper(IServiceConfig config)
             _env = Environment.GetEnvironmentVariable("ENVIRONMENT_NAME") ?? "Local";
         }
 
-        public IServiceConfigSection this[string name] => new Section(_config[name], _config[$"{name}-{_env}"]);
+        public IServiceConfigSection? this[string name] => new Section(_config[name], _config[$"{name}-{_env}"]);
```
Comment (ChadNedzlek) [inline src/Microsoft.DotNet.ServiceFabric.ServiceHost/EnvironmentConfigMapper.cs]: Why is this ?. It literally returns a new thing?

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.ServiceFabric.ServiceHost/EnvironmentConfigMapper.cs]: I hate nullable so much.  I really don't want us to use it, and I don't like that you are spreading it all over the code base with an unrelated change.

Comment (alexperovich) [reply src/Microsoft.DotNet.ServiceFabric.ServiceHost/EnvironmentConfigMapper.cs]: This is ? because the interface is, i can remove it though.

**Comment by alexperovich** [comment]
Yea, I ran the scenario tests on my local cluster

File: src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj [hunk 2]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <IsPackable>true</IsPackable>
     <TargetFrameworks>netcoreapp3.1;netcoreapp2.1</TargetFrameworks>
+    <Nullable>enable</Nullable>
```
Comment (ChadNedzlek) [inline src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj]: Why enable?  You keep slipping nullable into weird places...

Comment (ChadNedzlek) [reply src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj]: Especially when the target framework is 2.1, this doesn't make a lot of sense.

Comment (alexperovich) [reply src/Microsoft.DncEng.Configuration.Extensions/Microsoft.DncEng.Configuration.Extensions.csproj]: This project already had it enabled..... I am not going to remove them all from this one.

**Comment by alexperovich** [comment]
nullable was already enabled for those projects, I just removed it from the service host project.

**Comment by ChadNedzlek** [comment]
Ah, I see.

**Pull request merged**